### PR TITLE
[Don't merge ] Download course progress report

### PR DESCRIPTION
**Pull request by nilesh-more**

### CODE REVIEW

File: src/app/private/config/privateErrorMessages.json [hunk 1]
```diff
@@ -251,7 +251,8 @@
                     "editingProfile": "Updating profile ...",
                     "profileEdited": "Profile successfully updated"
                 },
-                "UPDATE": { 
+
```
Comment (kochhar) [inline src/app/private/config/privateErrorMessages.json]: don't need extra blank line.

File: src/app/private/scripts/controllers/dashboard/courseDashboardController.js [hunk 2]
```diff
@@ -198,6 +198,39 @@ angular.module('playerApp')
...
+        courseDashboard.showDownloadLoader = 'active';
+        var req = {
+          courseId: courseDashboard.batchIdentifier,
+          timePeriod: courseDashboard.filterTimePeriod
+        };
+
+        // Call service
+        dashboardService.downloadReport(req).then(function(apiResponse) {
+          courseDashboard.showDownloadLoader = '';
+          if (apiResponse && apiResponse.responseCode === 'OK') {
```
Comment (kochhar) [inline src/app/private/scripts/controllers/dashboard/courseDashboardController.js]: please push the request error handling into the service layer.

File: src/app/private/scripts/controllers/dashboard/courseDashboardController.js [hunk 3]
```diff
@@ -198,6 +198,39 @@ angular.module('playerApp')
...
+            var str = $rootScope.errorMessages.DASHBOARD.DOWNLOAD_REPORTS.COURSES_CSV_MSG;
+            courseDashboard.downloadReportText = str.replace("{acknowledgementId}", apiResponse.result.requestId).replace(/(\(.*\))/g, '');
+            $timeout(function() {
```
Comment (kochhar) [inline src/app/private/scripts/controllers/dashboard/courseDashboardController.js]: why is a timeout required here?

File: src/app/private/scripts/controllers/dashboard/courseDashboardController.js [hunk 4]
```diff
@@ -198,6 +198,39 @@ angular.module('playerApp')
...
+          courseId: courseDashboard.batchIdentifier,
+          timePeriod: courseDashboard.filterTimePeriod
+        };
+
+        // Call service
+        dashboardService.downloadReport(req).then(function(apiResponse) {
+          courseDashboard.showDownloadLoader = '';
+          if (apiResponse && apiResponse.responseCode === 'OK') {
+            var str = $rootScope.errorMessages.DASHBOARD.DOWNLOAD_REPORTS.COURSES_CSV_MSG;
+            courseDashboard.downloadReportText = str.replace("{acknowledgementId}", apiResponse.result.requestId).replace(/(\(.*\))/g, '');
```
Comment (kochhar) [inline src/app/private/scripts/controllers/dashboard/courseDashboardController.js]: why is a regexp replacement required on the downloaded text?

File: src/app/private/scripts/controllers/dashboard/orgDashboardController.js [hunk 5]
```diff
@@ -224,5 +224,39 @@ angular.module('playerApp')
...
+        dashboardData.showDownloadLoader = 'active';
+        var req = {
+          org_id: dashboardData.orgId,
+          period: dashboardData.timePeriod
+        };
+
+        // Call service
+        dashboardService.downloadReport(req, dashboardData.datasetPreviousValue).then(function(apiResponse) {
+          dashboardData.showDownloadLoader = '';
+          if (apiResponse && apiResponse.responseCode === 'OK') {
```
Comment (kochhar) [inline src/app/private/scripts/controllers/dashboard/orgDashboardController.js]: please push this error handling down into the service class.

File: src/app/private/scripts/controllers/dashboard/orgDashboardController.js [hunk 6]
```diff
@@ -224,5 +224,39 @@ angular.module('playerApp')
...
+       */
+      dashboardData.downloadReport = function() {
+        dashboardData.showDownloadLoader = 'active';
+        var req = {
+          org_id: dashboardData.orgId,
+          period: dashboardData.timePeriod
+        };
+
+        // Call service
+        dashboardService.downloadReport(req, dashboardData.datasetPreviousValue).then(function(apiResponse) {
```
Comment (kochhar) [inline src/app/private/scripts/controllers/dashboard/orgDashboardController.js]: this code block seems to be a duplicate of the code in course dashboard controller. please remove the copied code and use a shared implementation.

**Comment by harishgilimi** [comment]
@nilesh-more  move this changes to release-1.3 and close this PR

**Comment by harishgilimi** [comment]
closing PR for cleaning repo

**Pull request closed without merging**